From 4a76667680dcef4016561ccd6ef2f2e280fa63f8 Mon Sep 17 00:00:00 2001
From: LaserEyess <lasereyess@users.noreply.github.com>
Date: Mon, 15 Jun 2020 15:10:18 -0400
Subject: [PATCH] fix constructor ordering

gcc gives `-Wreorder` for FFT3DFilter because the order of the
constructor is wrong. Fix this by reordering the members and
attributes as they are in FFT3DFilter.h.
---
 FFT3DFilter.cpp | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/FFT3DFilter.cpp b/FFT3DFilter.cpp
index 6882c68..ad72238 100644
--- a/FFT3DFilter.cpp
+++ b/FFT3DFilter.cpp
@@ -193,12 +193,13 @@ kratio(_kratio), sharpen(_sharpen), scutoff(_scutoff), svr(_svr), smin(_smin), s
 measure(_measure), interlaced(_interlaced), wintype(_wintype),
 pframe(_pframe), px(_px), py(_py), pshow(_pshow), pcutoff(_pcutoff), pfactor(_pfactor),
 sigma2(_sigma2), sigma3(_sigma3), sigma4(_sigma4), degrid(_degrid),
-dehalo(_dehalo), hr(_hr), ht(_ht), ncpu(_ncpu),
-vi(_vi), node(_node), wsharpen(nullptr, nullptr), wdehalo(nullptr, nullptr),
-pattern2d(nullptr, nullptr), pattern3d(nullptr, nullptr), in(nullptr, nullptr),
-gridsample(nullptr, nullptr), outLast(nullptr, nullptr), covar(nullptr, nullptr),
-covarProcess(nullptr, nullptr), outrez(nullptr, nullptr), plan(nullptr, nullptr), planinv(nullptr, nullptr),
-plan1(nullptr, nullptr) {
+dehalo(_dehalo), hr(_hr), ht(_ht), ncpu(_ncpu), in(nullptr, nullptr),
+outrez(nullptr, nullptr), gridsample(nullptr, nullptr), plan(nullptr, nullptr),
+planinv(nullptr, nullptr), plan1(nullptr, nullptr),
+wsharpen(nullptr, nullptr), wdehalo(nullptr, nullptr),
+outLast(nullptr, nullptr), covar(nullptr, nullptr),
+covarProcess(nullptr, nullptr), pattern2d(nullptr, nullptr),
+pattern3d(nullptr, nullptr), vi(_vi), node(_node) {
     int istat = fftwf_init_threads();
     if (istat == 0)
         throw std::runtime_error{ "fftwf_init_threads() failed!" };
